pandas/core/generic.py

```diff
@@ -195,9 +195,12 @@ def __unicode__(self):
         return '%s(%s)' % (self.__class__.__name__, prepr)
 
     def _dir_additions(self):
-        """ add the string-like attributes from the info_axis """
-        additions = set([c for c in self._info_axis
-                         if isinstance(c, string_types) and isidentifier(c)])
+        """ add the string-like attributes from the info_axis.
+        If info_axis is a MultiIndex, it's first level values are used.
+        """
+        additions = set(
```

Comment:
can you add an asv that does dir() on a Series/DataFrame with say 10000 elements (we *might* already have one of these)